homeassistant/util/package.py

```diff
@@ -96,18 +103,43 @@ def install_package(
     """
     _LOGGER.info("Attempting install of %s", package)
     env = os.environ.copy()
-    args = ["uv", "pip", "install", "--quiet", package]
+    args = [
+        "uv",
+        "pip",
+        "install",
+        "--quiet",
+        package,
+        # We need to use unsafe-first-match for custom components
+        # which can use a different version of a package than the one
+        # we have built the wheel for.
+        "--index-strategy",
+        "unsafe-first-match",
+    ]
     if timeout:
         env["HTTP_TIMEOUT"] = str(timeout)
     if upgrade:
         args.append("--upgrade")
     if constraints is not None:
         args += ["--constraint", constraints]
+    not_venv = not is_virtual_env()
     if target:
-        assert not is_virtual_env()
+        assert not_venv
```

Comment:
Why are we using assert here? It doesn't look like just a type annotation hint. If we need to catch runtime errors we shouldn't use assert since it can be disabled.

Reply:
The assert was already here but I will change it